### PR TITLE
Add isPlainObject

### LINKED ISSUE #119: isPlainObject
```
		expect(isPlainObject(null)).toBe(false);
		expect(isPlainObject([])).toBe(false);
		expect(isPlainObject(new Date())).toBe(false);
		expect(isPlainObject(new Error())).toBe(false);
		expect(isPlainObject(undefined)).toBe(false);
		expect(isPlainObject('[object Object]')).toBe(false);

		expect(isPlainObject({})).toBe(true);
		expect(isPlainObject(Object.create(null))).toBe(true);
		expect(isPlainObject({ constructor: () => {} })).toBe(true);
```

### PR DESCRIPTION
**Pull request by wafflepie**
Closes #119

### CODE REVIEW

File: packages/ramda-extension/src/isPlainObject.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * @example
+ *
+ *        R_.isPlainObject({}) // true
+ *        R_.isPlainObject([]) // false
+ *        R_.isPlainObject(null) // false
+ *
+ * @sig a -> Boolean
+ */
+const isPlainObject = allPass([
+	o(equals('Object'), type),
```
Comment (tommmyy) [inline packages/ramda-extension/src/isPlainObject.js]: Can't you use `R_.isObject`?

File: packages/ramda-extension/src/isPlainObject.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	isNotNil,
+	either(
+		o(equals(Object.prototype), Object.getPrototypeOf),
+		// NOTE: prototype is null if created using Object.create(null)
+		o(isNil, Object.getPrototypeOf)
```
Comment (tommmyy) [inline packages/ramda-extension/src/isPlainObject.js]: is not there already check for nil in `isNotNil` on line 23?

Comment (wafflepie) [reply packages/ramda-extension/src/isPlainObject.js]: This is to check the prototype of the object.
`[[Prototype]]` of `Object.create(null)` is `null`, whereas `[[Prototype]]` of `{}` is `Object.prototype`

File: packages/ramda-extension/src/isPlainObject.js (hunk 1 above)
Comment (wafflepie) [reply packages/ramda-extension/src/isPlainObject.js]: `Object.create(null) instanceof Object // false` :\

**Comment by tommmyy** [comment]
@wafflepie thank you for the great work!

**Pull request merged**